exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/CryptoFunctions.java

```diff
@@ -0,0 +1,315 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.expr.fn.impl;
+
+import io.netty.buffer.DrillBuf;
+import org.apache.drill.exec.expr.DrillSimpleFunc;
+import org.apache.drill.exec.expr.annotations.FunctionTemplate;
+import org.apache.drill.exec.expr.annotations.Output;
+import org.apache.drill.exec.expr.annotations.Param;
+import org.apache.drill.exec.expr.annotations.Workspace;
+import org.apache.drill.exec.expr.holders.VarCharHolder;
+
+import javax.inject.Inject;
+
+public class CryptoFunctions{
+    static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(CryptoFunctions.class);
+
+    private CryptoFunctions() {}
+
+    @FunctionTemplate(
+        name = "md5",
+        scope = FunctionTemplate.FunctionScope.SIMPLE,
+        nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+    )
+    public static class md5Function implements DrillSimpleFunc {
+
+        @Param
+        VarCharHolder raw_input;
+
+        @Output
+        VarCharHolder out;
+
+        @Inject
+        DrillBuf buffer;
+
+        @Workspace
+        java.security.MessageDigest md;
+
+        @Override
+        public void setup() {
+            try {
+                md = java.security.MessageDigest.getInstance("MD5");
+            } catch( Exception e ) {
+            }
+        }
+
+        @Override
+        public void eval() {
+
+            String input = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(raw_input.start, raw_input.end, raw_input.buffer);
+            byte[] thedigest = null;
+            String output_string = "";
+
+            try {
+                byte[] bytesOfMessage = input.getBytes("UTF-8");
+                thedigest = md.digest(bytesOfMessage);
+                output_string = String.format("%032X", new java.math.BigInteger(1, thedigest));
+                output_string = output_string.toLowerCase();
+
+            } catch( Exception e ) {
+            }
```

Comment:
1. Please mind formatting.
2. If in case of error, we leave the defaults, then it's better  to at least log the exception.